new_build_defs.bzl

```diff
@@ -0,0 +1,248 @@
+# Copyright 2024 Figure AI, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Improved bazel build rules for flatbuffers.
+"""
+
+load("@rules_cc//cc:defs.bzl", "cc_library")
+load("@rules_python//python:defs.bzl", "py_library")
+
+DEFAULT_FLATC_ARGS = [
+    "--cpp-ptr-type flatbuffers::unique_ptr",
```

Comment:
We provide our own ptr type in our rules:
```
# FBS Pointer type for objects, lazy vs default initialization
OBJ_LAZY_UNIQ_PTR = "figure::LazyUniquePtr"
OBJ_INSTANTIATED_UNIQ_PTR = "figure::InstantiatedUniquePtr"
```

Reply:
ditto, keeping these internal.